<a href="https://colab.research.google.com/github/HeatherAnnFoster/Regression--Prediciton-of-Grocery-Sales/blob/main/P_1_W2_Heather_Foster_Regression_Prediction_of_Grocery_Sales.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Regression- Predicition of Grocery Sales#
Heather Foster


## Loading Data ##


In [76]:
import pandas as pd


In [77]:
df = pd.read_excel('/content/sales_predictions.xlsx')

In [78]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


In [79]:
df.head()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15         9.30          Low Fat         0.016047   
1           DRC01         5.92          Regular         0.019278   
2           FDN15        17.50          Low Fat         0.016760   
3           FDX07        19.20          Regular         0.000000   
4           NCD19         8.93          Low Fat         0.000000   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   

         Outlet_Type  Item_Outlet_Sales  
0  Supermarket Type1          3735.1380  
1  Supermarket Type2           443.4228  
2  Supermarket Type1          2097.2700  
3      Grocery Store           732.3800  
4  Supermarket Type1           994.7052

In [80]:
df.tail()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
8518           FDF22        6.865          Low Fat         0.056783   
8519           FDS36        8.380          Regular         0.046982   
8520           NCJ29       10.600          Low Fat         0.035186   
8521           FDN46        7.210          Regular         0.145221   
8522           DRG01       14.800          Low Fat         0.044878   

               Item_Type  Item_MRP Outlet_Identifier  \
8518         Snack Foods  214.5218            OUT013   
8519        Baking Goods  108.1570            OUT045   
8520  Health and Hygiene   85.1224            OUT035   
8521         Snack Foods  103.1332            OUT018   
8522         Soft Drinks   75.4670            OUT046   

      Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
8518                       1987        High               Tier 3   
8519                       2002         NaN               Tier 2   
8520                       2004       Small               Tier 2   
8521                       2009      Medium               Tier 3   
8522                       1997       Small               Tier 1   

            Outlet_Type  Item_Outlet_Sales  
8518  Supermarket Type1          2778.3834  
8519  Supermarket Type1           549.2850  
8520  Supermarket Type1          1193.1136  
8521  Supermarket Type2          1845.5976  
8522  Supermarket Type1           765.6700

## Data Cleaning ##

In [81]:

df.shape

(8523, 12)

There are 8,523 columns and 12 rows in the dataset.

In [82]:

df.dtypes

Item_Identifier               object
Item_Weight                  float64
Item_Fat_Content              object
Item_Visibility              float64
Item_Type                     object
Item_MRP                     float64
Outlet_Identifier             object
Outlet_Establishment_Year      int64
Outlet_Size                   object
Outlet_Location_Type          object
Outlet_Type                   object
Item_Outlet_Sales            float64
dtype: object

The Data types are listed above.

In [83]:
df.duplicated().sum()

0

In [84]:

df.isna().sum()

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

There are signiciant missing values from Item_Weight and Outlet _size.  Everything else seems to be accounted for.

In [85]:

median_Item_Weight=df['Item_Weight'].median()
df['Item_Weight'].median()


12.6

There is too much data to drop, in terms of calues for each of the columns.  In order to have a representative look at the values of these columns, I will be filling in the missing data with the median value of the date.  The reason I will be using the median is because this number is less affected by the outliers within the dataset and will produce less bias moving forward.  The median value for Item_Weight is 12.6.

In [86]:
df['Item_Weight']

0        9.300
1        5.920
2       17.500
3       19.200
4        8.930
         ...  
8518     6.865
8519     8.380
8520    10.600
8521     7.210
8522    14.800
Name: Item_Weight, Length: 8523, dtype: float64

In [87]:
df['Item_Weight'].fillna('Missing', inplace = True)
df.isna().sum()


Item_Identifier                 0
Item_Weight                     0
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

The average for the Item_Weight has been changed to 12.6 and inputted into the dataset.  Therefore, the Item_Weight now has 0 missing values.

The next column to be fixed is the Outlet_Size column.  It is missing 2,410 values.  I will be inputting the median value of this column to complete the data because I would lose too much information if I deleted the column all togther.

In [88]:
df['Outlet_Size']

0       Medium
1       Medium
2       Medium
3          NaN
4         High
         ...  
8518      High
8519       NaN
8520     Small
8521    Medium
8522     Small
Name: Outlet_Size, Length: 8523, dtype: object

The Outlet_Size was changed to the size that was most ordered.

In [89]:
df['Outlet_Size'].fillna('Missing', inplace = True)
df.isna().sum()

Item_Identifier              0
Item_Weight                  0
Item_Fat_Content             0
Item_Visibility              0
Item_Type                    0
Item_MRP                     0
Outlet_Identifier            0
Outlet_Establishment_Year    0
Outlet_Size                  0
Outlet_Location_Type         0
Outlet_Type                  0
Item_Outlet_Sales            0
dtype: int64

In [90]:
df.isna().sum()

Item_Identifier              0
Item_Weight                  0
Item_Fat_Content             0
Item_Visibility              0
Item_Type                    0
Item_MRP                     0
Outlet_Identifier            0
Outlet_Establishment_Year    0
Outlet_Size                  0
Outlet_Location_Type         0
Outlet_Type                  0
Item_Outlet_Sales            0
dtype: int64

This is the double check to make sure no columns are missing values.  Because everything has a 0 value in it, this shows that all the columns and rows have a value in it.

In [100]:
dict = {"low fat": "Low Fat", "reg": "Regular", "LF": "Low Fat"}
df.replace({'Item_Fat_Content':dict}, inplace = True)

The inconsistencies have been replaced and fixed.  They are uniform throughout the dataset.  The following excerpt of data shows the change.

In [103]:
df.head(30)

Item_Identifier Item_Weight Item_Fat_Content  Item_Visibility  \
0            FDA15         9.3          Low Fat         0.016047   
1            DRC01        5.92          Regular         0.019278   
2            FDN15        17.5          Low Fat         0.016760   
3            FDX07        19.2          Regular         0.000000   
4            NCD19        8.93          Low Fat         0.000000   
5            FDP36      10.395          Regular         0.000000   
6            FDO10       13.65          Regular         0.012741   
7            FDP10     Missing          Low Fat         0.127470   
8            FDH17        16.2          Regular         0.016687   
9            FDU28        19.2          Regular         0.094450   
10           FDY07        11.8          Low Fat         0.000000   
11           FDA03        18.5          Regular         0.045464   
12           FDX32        15.1          Regular         0.100014   
13           FDS46        17.6          Regular         0.047257   
14           FDF32       16.35          Low Fat         0.068024   
15           FDP49         9.0          Regular         0.069089   
16           NCB42        11.8          Low Fat         0.008596   
17           FDP49         9.0          Regular         0.069196   
18           DRI11     Missing          Low Fat         0.034238   
19           FDU02       13.35          Low Fat         0.102492   
20           FDN22       18.85          Regular         0.138190   
21           FDW12     Missing          Regular         0.035400   
22           NCB30        14.6          Low Fat         0.025698   
23           FDC37     Missing          Low Fat         0.057557   
24           FDR28       13.85          Regular         0.025896   
25           NCD06        13.0          Low Fat         0.099887   
26           FDV10       7.645          Regular         0.066693   
27           DRJ59       11.65          Low Fat         0.019356   
28           FDE51       5.925          Regular         0.161467   
29           FDC14     Missing          Regular         0.072222   

                Item_Type  Item_MRP Outlet_Identifier  \
0                   Dairy  249.8092            OUT049   
1             Soft Drinks   48.2692            OUT018   
2                    Meat  141.6180            OUT049   
3   Fruits and Vegetables  182.0950            OUT010   
4               Household   53.8614            OUT013   
5            Baking Goods   51.4008            OUT018   
6             Snack Foods   57.6588            OUT013   
7             Snack Foods  107.7622            OUT027   
8            Frozen Foods   96.9726            OUT045   
9            Frozen Foods  187.8214            OUT017   
10  Fruits and Vegetables   45.5402            OUT049   
11                  Dairy  144.1102            OUT046   
12  Fruits and Vegetables  145.4786            OUT049   
13            Snack Foods  119.6782            OUT046   
14  Fruits and Vegetables  196.4426            OUT013   
15              Breakfast   56.3614            OUT046   
16     Health and Hygiene  115.3492            OUT018   
17              Breakfast   54.3614            OUT049   
18            Hard Drinks  113.2834            OUT027   
19                  Dairy  230.5352            OUT035   
20            Snack Foods  250.8724            OUT013   
21           Baking Goods  144.5444            OUT027   
22              Household  196.5084            OUT035   
23           Baking Goods  107.6938            OUT019   
24           Frozen Foods  165.0210            OUT046   
25              Household   45.9060            OUT017   
26            Snack Foods   42.3112            OUT035   
27            Hard Drinks   39.1164            OUT013   
28                  Dairy   45.5086            OUT010   
29                 Canned   43.6454            OUT019   

    Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                        1999      Medium               Tier

## Exploratory Visuals ##

## Exploratory Visuals ##